packages/forma-36-react-components/src/components/Dropdown/DropdownContainer/DropdownContainer.tsx

```diff
@@ -1,252 +1,131 @@
-        'bottom-right': 'bottom-left',
-        'top-right': 'top-left',
-        left: 'right',
-      },
-      top: {
-        'top-left': 'bottom-left',
-        'top-right': 'bottom-right',
+    refCallback,
+  ) => {
+    // We're not dealing with React RefObjects but with useState (because we
+    // want to re-render on all changes)
+    const setReference = refCallback as React.Dispatch<
+      React.SetStateAction<HTMLElement | null>
+    >;
+    const dropdown = useRef<HTMLDivElement | null>(null);
+    const portalTarget = useRef<HTMLDivElement>(document.createElement('div'));
+    const classNames = cn(className, styles['DropdownContainer']);
+
+    const trackOutsideClick = useCallback(
+      (event: MouseEvent) => {
+        if (
+          isOpen &&
+          onClose &&
+          dropdown.current &&
+          !dropdown.current.contains(event.target as Node)
+        ) {
+          onClose();
+        }
       },
-      bottom: {
-        'bottom-left': 'top-left',
-        'bottom-right': 'top-right',
-      },
-    };
-    const currentPosition = this.state.position;
-    const resolution = resolutions[overflowAt][currentPosition];
-    if (resolution) {
-      this.setState(
-        {
-          position: resolution,
-        },
-        () => {
-          this.lastOverflowAt = overflowAt;
-        },
-      );
-    }
-  };
-
-  calculatePosition = () => {
-    const { anchorDimensionsAndPositon } = this.props;
-    const { dropdownDimensions, position } = this.state;
-
-    if (!anchorDimensionsAndPositon || !dropdownDimensions) {
-      return false;
-    }
-
-    switch (position) {
-      case 'bottom-left':
-        return {
-          top:
-            anchorDimensionsAndPositon.top + anchorDimensionsAndPositon.height,
-          left: anchorDimensionsAndPositon.left,
-        };
-      case 'top-left':
-        return {
-          bottom: window.innerHeight - anchorDimensionsAndPositon.top,
-          left: anchorDimensionsAndPositon.left,
-        };
-      case 'bottom-right':
-        return {
-          top:
-            anchorDimensionsAndPositon.top + anchorDimensionsAndPositon.height,
-          left:
-            anchorDimensionsAndPositon.left -
-            (dropdownDimensions.width - anchorDimensionsAndPositon.width),
-        };
-      case 'top-right':
-        return {
-          bottom: window.innerHeight - anchorDimensionsAndPositon.top,
-          left:
-            anchorDimensionsAndPositon.left -
-            (dropdownDimensions.width - anchorDimensionsAndPositon.width),
-        };
-    }
-  };
-
-  getSubmenuClassNames = () =>
-    cn(
-      styles['DropdownContainer__submenu'],
-      styles[`DropdownContainer__container-position--${this.state.position}`],
+      [isOpen, onClose, dropdown.current],
     );
 
-  render() {
-    const { submenu, className, width, testId } = this.props;
+    useEffect(() => {
+      if (isOpen) {
+        document.body.appendChild(portalTarget.current);
```

Comment:
@denkristoffer if you can find a way to not to use portal it would be awesome. the problem with portal is keyboard focus order; upon open the dropdown content is at the end of the document, onClose the next keyboard element to be focused is the first element in the document. I hope that makes any sense, let's talk about it if not :)

Reply:
@mshaaban0 I agree, the portal should be removed by default. I think we should handle this separately in https://github.com/contentful/forma-36/issues/601 though, also to make sure this PR doesn't break any existing uses of the dropdown. Is that okay with you?